### PR TITLE
u2f: Enable tests without hardware

### PR DESCRIPTION
**Pull request by NickeZ**
This PR hooks up the `hid_read`/`hid_write` to internal functions to simulate a connected device.

### CODE REVIEW

File: src/queue.h [hunk 1]
```diff
@@ -37,4 +37,6 @@ uint8_t* queue_pull(void);
  */
 void queue_clear(void);
 
+uint8_t* queue_peek(void);
```
Comment (benma) [inline src/queue.h]: can the return be made const?

File: test/unit-test/framework/mock_securechip.c [hunk 2]
```diff
@@ -36,3 +38,19 @@ bool securechip_kdf(securechip_slot_t slot, const uint8_t* msg, size_t len, uint
...
+
+bool securechip_u2f_counter_inc(uint32_t* counter)
+{
+    *counter = _u2f_counter++;
+    return true;
+}
+
+bool securechip_ecc_unsafe_sign(const uint8_t* priv_key, const uint8_t* msg, uint8_t* sig)
+{
+    return false;
```
Comment (benma) [inline test/unit-test/framework/mock_securechip.c]: do the tests not exercise signing?

Comment (NickeZ) [reply test/unit-test/framework/mock_securechip.c]: not right now, I guess I could mock it somehow, but I can wait with that.

File: src/queue.h (hunk 1 above)
Comment (NickeZ) [reply src/queue.h]: fixed

**Pull request merged**